### PR TITLE
Make server and remote options as required

### PR DESCRIPTION
**Pull request by pleum**
Make server and remote options as required
fix: https://github.com/davrodpin/mole/issues/6

### CODE REVIEW

**Comment by davrodpin** [comment]
Hi @pleum,

Thank you very much for contributing to the project!

How about having a new method on `cli.App` to validate the arguments based on the parsed command? Something like:

Add a new `validate` method to `cli.App` to validate the given options based on the parsed command:

```go
func (a *App) Validate() (valid bool, errors []string) {

}
```

**Comment by pleum** [comment]
@davrodpin Sorry i'm still new for working open source and just wondering why need to return `bool` while if errors is not `nil` that should raise error anyway?

**Comment by davrodpin** [comment]
> @davrodpin Sorry i'm still new for working open source and just wondering why need to return `bool` while if errors is not `nil` that should raise error anyway?

@pleum, don't worry about being new to open source. This is my first open source project, so we are even 😄 

And you're right, the code snippet I provided looks too complicated and not communicating the message well.

How about a simple `Validate` method that will return the first validation error found or nil if there is no validation issues?

I also found a library that makes this implementation easier: https://github.com/go-ozzo/ozzo-validation.  

There is only one use case to be covered right now, but once this code infra is in place, it will make way easier to add other validations in the future.

File: cli/cli.go [hunk 1]
```diff
@@ -67,6 +70,21 @@ func (c *App) Parse() error {
...
 
+	err := c.Validate()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c App) Validate() error {
```
Comment (davrodpin) [inline cli/cli.go]: You will be able to use `.Required` if you change the receiver to be a pointer.

```suggestion
func (c *App) Validate() error {
```

File: cli/cli.go [hunk 2]
```diff
@@ -1,10 +1,13 @@
 package cli
 
 import (
+	"errors"
 	"flag"
 	"fmt"
 	"regexp"
 	"strings"
+
+	"github.com/go-ozzo/ozzo-validation"
```
Comment (davrodpin) [inline cli/cli.go]: Please add the dependency to the `vendor/` folder.

File: cli/cli.go (hunk 1 above)
Comment (pleum) [reply cli/cli.go]: It give me error  `only a pointer to a struct can be validated`

File: cli/cli.go (hunk 2 above)
Comment (davrodpin) [reply cli/cli.go]: The code below is working for me:

```go
func (c *App) Validate() error {
  if c.Command == "new-alias" {
    return validation.ValidateStruct(c,
      validation.Field(c.Server, validation.Required),
      validation.Field(c.Remote, validation.Required))
  }

  return nil
}
```

Comment (pleum) [reply cli/cli.go]: But it doesn't work for me:
err

Comment (davrodpin) [reply cli/cli.go]: Could you check what is the latest commit of `github.com/go-ozzo/ozzo-validation` that you have to your Go workspace? Mine is pointing to `106681dbb37bfa3e7683c4c8129cb7f5925ea3e9` related to tag `v3.5.0`.

Comment (pleum) [reply cli/cli.go]: img
and here is my `vendor.json`
```
{
  "checksumSHA1": "lZHmhkz+QdNpbGMC05esL0bg3Gs=",
  "path": "github.com/go-ozzo/ozzo-validation",
  "revision": "106681dbb37bfa3e7683c4c8129cb7f5925ea3e9",
  "revisionTime": "2018-09-29T20:53:26Z"
}
```

Comment (davrodpin) [reply cli/cli.go]: I don't know what `vendor.js` is for, but how about proceed with the implementation  without using  `ozzo-validation`?

**Comment by davrodpin** [comment]
@pleum, how about adding a test for the `Validate` method?

**Comment by davrodpin** [comment]
Thanks for you contribution!

**Pull request merged**